src/codegen/codegen_neuron_cpp_visitor.cpp

```diff
@@ -405,6 +410,73 @@ void CodegenNeuronCppVisitor::print_mechanism_global_var_structure(bool print_in
                       info.pointer_variables.size() > 0
                           ? static_cast<int>(info.pointer_variables.size())
                           : -1);
+
+    // Start printing the CNRN-style global variables.
+    auto float_type = default_float_data_type();
+    printer->add_newline(2);
+    printer->add_line("/** all global variables */");
+    printer->fmt_push_block("struct {}", global_struct());
+
+    if (!info.ions.empty()) {
+        // TODO implement these when needed.
+    }
+
+    if (info.point_process) {
+        throw std::runtime_error("Not implemented, global point process.");
+    }
+
+    if (!info.vectorize && !info.top_local_variables.empty()) {
+        throw std::runtime_error("Not implemented, global vectorize something.");
+    }
+
+    if (!info.thread_variables.empty()) {
+        throw std::runtime_error("Not implemented, global thread variables.");
+    }
+
+    if (info.table_count > 0) {
+        throw std::runtime_error("Not implemented, global table count.");
+    }
+
+    for (const auto& var: info.state_vars) {
+        auto name = var->get_name() + "0";
+        auto symbol = program_symtab->lookup(name);
+        if (symbol == nullptr) {
+            printer->fmt_line("{} {}{};", float_type, name, value_initialize);
+            codegen_global_variables.push_back(make_symbol(name));
+        }
+    }
+
+    for (const auto& var: info.global_variables) {
+        auto name = var->get_name();
+        auto length = var->get_length();
+        if (var->is_array()) {
```

Comment:
I think we're missing the proper implementation for this in the CoreNEURON codegen as well. Could you create an issue about this? Could be a `good first issue` for newcomers

Reply:
What is "this"? Adding an initializer for a const array?

Reply:
Yes that's what I meant. However looking at this again I am not sure what the use case that isn't working. I can't find how a `GLOBAL` array can be const or initialized in a MOD file except inside a `FUNCTION/PROCEDURE/INITIAL/STATE/BREAKPOINT` block which should work properly.
I saw that the `TODO` was added in https://github.com/BlueBrain/nmodl/pull/904. I am wondering whether @pramodk has any idea what's missing?

Reply:
I wasn't able to guess the correct syntax for global array variables. Meaning I tried a couple of sensible looking ideas against NRN + nocmodl, but they didn't work.